ui/src/app/pages/profile/profile-component.tsx

```diff
@@ -552,12 +552,29 @@ export const ProfileComponent = fp.flow(
                 <div style={{ marginLeft: '1.5rem' }}>
                   <div style={styles.title}>Initial credits balance</div>
                   <hr style={{ ...styles.verticalLine }} />
-                  {profile && (
+                  {currentProfile && (
                     <InitialCreditsPanel
-                      freeTierUsage={profile.freeTierUsage}
-                      freeTierDollarQuota={profile.freeTierDollarQuota}
+                      freeTierUsage={currentProfile.freeTierUsage}
+                      freeTierDollarQuota={currentProfile.freeTierDollarQuota}
                       expirationDate={
-                        profile.initialCreditsExpirationEpochMillis
+                        currentProfile.initialCreditsExpirationEpochMillis
+                      }
+                      eligibleForExtension={
+                        currentProfile.eligibleForInitialCreditsExtension
+                      }
+                      updateInitialCredits={(
+                        updatedExpiration: number,
+                        updatedExtensionEligibility: boolean
+                      ) =>
+                        this.setState({
```

Comment:
This updates expiration time and extension eligibility to avoid calling the profile endpoint again.